lib/immobilienscout/request.rb

```diff
@@ -12,13 +12,13 @@ def initialize(url, params = nil)
 
     def get
       auth_header = generate_auth_header('GET', @params)
-      headers = generate_headers(auth_header)
 
       url_with_query = "#{@url}?#{@params.to_query}"
       uri = URI.parse(url_with_query)
 
       request = Net::HTTP::Get.new(uri)
-      request.initialize_http_header(headers)
+      request.add_field('Authorization', auth_header)
+      request.add_field('Accept', 'application/xml')
 
       execute_request(uri, request)
```

Comment:
Why did you make this change for _all_ GET requests? We still want to call other endpoints via JSON, don't we?

I would have assumed that we add a separate entity for making XML requests. E.g. a base class `Request::Base` and then two subclasses `Request::JSON` and `Request::XML` which implement their respective means of generating headers and parsing the response.

Or, even better IMO, we could use the strategy pattern to inject this kind of behavior. The default strategy could be JSON and the one specific report request could override it to XML.

Reply:
I think it's because we use the GET request only for the scout report so far. But yes, if we want to implement other features at some point, it's probably cleaner/more reusable if we can use XML/JSON with all types of requests.

Reply:
Yes, @wjfrey is right is for that reason, so far we only have `ScoutReport` using get

Reply:
I understand that, I just think it's the wrong place/abstraction to perform this fix.